db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/task/schedule/CompositeParser.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) Gustav Karlsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.kagkarlsson.scheduler.task.schedule;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+final class CompositeParser implements Parser {
+    private final List<Parser> delegates;
+
+    private CompositeParser(List<Parser> delegates) {
+        this.delegates = delegates;
+    }
+
+    @Override
+    public Optional<Schedule> parse(String scheduleString) {
+        return delegates.stream()
+            .map(NotThrowingParser::of)
```

Comment:
Not sure if this is required, I thought we should still guard against parser implementations that throw exceptions.

Reply:
As we discussed elsewhere, maybe the strategy should be:
* If the regex does not match => Optional.empty()
* If the regex matches, but some extracted field on wrong format => runtimeexception

And in this case, maybe the best thing is to let it be uncaught so it hits the user, i.e. we do not need the `NotThrowingParser`. Do you agree?

Reply:
Yes, let's do this :+1: